lib/primer/forms/primer_text_field.ts

```diff
@@ -0,0 +1,42 @@
+import '@github/auto-check-element'
+import {controller, target} from '@github/catalyst'
+
+@controller
+class PrimerTextFieldElement extends HTMLElement {
+  @target inputElement: HTMLInputElement
+  @target validationElement: HTMLElement
+  @target validationMessageElement: HTMLElement
+
+  connectedCallback(): void {
+    this.inputElement.addEventListener('auto-check-success', () => { this.clearError() })
+
+    this.inputElement.addEventListener('auto-check-error', (event: any) => {
+      event.detail.response.text().then((error_message: string) => {
+        this.setError(error_message)
+      })
+    })
+  }
```

Comment:
For robustness, here it might be good to use an [AbortSignal so events can be correctly torn down](https://catalyst.rocks/guide/patterns/#registering-global-or-many-event-listeners):

```suggestion
  #abortController: AbortController | null
  
  connectedCallback(): void {
    this.#abortController?.abort()
    const signal = (this.#abortController = new AbortController())
    this.inputElement.addEventListener('auto-check-success', () => { this.clearError() }, { signal })

    this.inputElement.addEventListener('auto-check-error', (event: any) => {
      event.detail.response.text().then((error_message: string) => {
        this.setError(error_message)
      })
    }, { signal })
  }
  
  disconnectedCallback() {
    this.#abortController.abort()
  }
```